src/Network/GRPC/Common/HTTP2Settings.hs

```diff
@@ -0,0 +1,95 @@
+-- | Settings and parameters pertaining to HTTP\/2
+--
+-- Intended for unqualified import.
+
+module Network.GRPC.Common.HTTP2Settings
+  ( HTTP2Settings(..)
+  , defaultHTTP2Settings
+  ) where
+
+import Data.Default
+import Data.Word
+
+-- | HTTP\/2 settings
+data HTTP2Settings = HTTP2Settings {
+      -- | Maximum number of concurrent active streams
+      --
+      -- <https://datatracker.ietf.org/doc/html/rfc7540#section-5.1.2>
+      http2MaxConcurrentStreams :: Word32
+
+      -- | Initial window size for streams
+      --
+      -- <https://datatracker.ietf.org/doc/html/rfc7540#section-6.9.2>
+    , http2InitialStreamWindowSize :: Word32
+
+      -- | Initial connection window size
+      --
+      -- This setting is specific to the
+      -- [http2](https://hackage.haskell.org/package/http2) package's
+      -- implementation of the HTTP\/2 specification. In particular, this value
+      -- will be broadcast via a @WINDOW_UDPATE@ frame at the beginning of a new
+      -- connection.
+      --
+      -- If the consumed window space of all streams exceeds this value, the
+      -- sender will stop sending data. Therefore, if this value is less than
+      -- @'http2MaxConcurrentStreams' * 'http2InitialStreamWindowSize'@, there
+      -- is risk of a control flow deadlock, since the connection window space
+      -- may be used up by streams that we are not yet processing before we have
+      -- received all data on the streams that we /are/ processing. To reduce
+      -- this risk, increase 'serverNumberOfWorkers'. See
```

Comment:
Do the haddocks for `serverNumberOfWorkers` get resolved correctly here? You probably need to use a fully qualified name?